classes/local/tablesql/unsuccessful_enrolments_table_sql.php

```diff
@@ -265,9 +265,15 @@ public function col_actions($values) {
                 );
                 $contactmergerequest = reset($contactmergerequests);
                 $sourcecontact = $contactmergerequest->get_source_contact();
-                $sourceenrolments = $sourcecontact->get_associated_user()->has_course_enrolments();
-                $destinationenrolments = $contact->get_associated_user()->has_course_enrolments();
-                if (!($sourceenrolments && $destinationenrolments)) {
+
+                // Merge failed errors are misbehaving and sometimes have no source contact.
+                // Minor patch while that is fixed.
+                if (!empty($sourcecontact)) {
+                    $sourceenrolments = $sourcecontact->get_associated_user()->has_course_enrolments();
+                    $destinationenrolments = $contact->get_associated_user()->has_course_enrolments();
+                }
```

Comment:
This is problematic as variables may end up being undefined so I would suggest to initialize them before checking for sourcecontact like this:

```php
                $sourceenrolments = false;
                $destinationenrolments = false;
                if (!empty($sourcecontact)) {
                    $sourceenrolments = $sourcecontact->get_associated_user()->has_course_enrolments();
                    $destinationenrolments = $contact->get_associated_user()->has_course_enrolments();
                }
```